### PR TITLE
Context aware alert analysis

### PR DESCRIPTION
**Pull request by songkant-aws**
### Description
Context aware alert analysis implementation in alerts dashboard by leveraging existing IncontextInsight 'GeneratePopover' component

In-context prompting for alerts entry point:
Screenshot 2024-07-12 at 14 28 18
In-context prompting generating process:
Screenshot 2024-07-12 at 14 28 23
In-context prompting showing summary result with option to continue the conversation in chat:
Screenshot 2024-07-12 at 14 28 06
 
### Issues Resolved
https://github.com/opensearch-project/alerting-dashboards-plugin/issues/995
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by xluo-aws** [comment]
Can some maintainers help review this ticket and add tag 2.17?

File: public/pages/Dashboard/containers/Dashboard.js
Comment (amsiglan) [inline public/pages/Dashboard/containers/Dashboard.js:470]: Why are we changing this to `== 0`? I don't think we want to enable selecting the alerts in this case since it will enable certain actions like Acknowledge

File: public/pages/Dashboard/utils/tableUtils.js
Comment (amsiglan) [inline public/pages/Dashboard/utils/tableUtils.js:231]: Nit: `: contextProvider` not needed

File: public/types.ts
Comment (amsiglan) [inline public/types.ts:12]: Why can't we import this directly where needed?

File: public/pages/Dashboard/utils/tableUtils.js
Comment (amsiglan) [inline public/pages/Dashboard/utils/tableUtils.js:161]: what if the get monitor fails for some reason?

File: public/pages/Dashboard/utils/tableUtils.js
Comment (amsiglan) [inline public/pages/Dashboard/utils/tableUtils.js:177]: What about the other monitor types?

File: public/pages/Home/Home.js
Comment (amsiglan) [inline public/pages/Home/Home.js:1]: the changes in this file seem redundant. Can we revert them?

**Review by amsiglan — APPROVED** [review]

**Review by amsiglan — APPROVED** [review]

**Review by amsiglan — APPROVED** [review]

**Review by riysaxen-amzn — APPROVED** [review]

**Comment by riysaxen-amzn** [comment]
let's wait for the CI to pass before merging

**Pull request merged**
